packages/protocol/contracts/common/MetaTransactionWallet.sol

```diff
@@ -47,6 +53,16 @@ contract MetaTransactionWallet is
     bytes returnData
   );
 
+  event MetaTransactionWithRefundExecution(
+    address indexed destination,
+    uint256 value,
+    bytes data,
+    uint256 indexed nonce,
+    uint256 maxGasPrice,
+    uint256 metaGasLimit,
+    bytes returnData
```

Comment:
One idea to simplify the events would be to just add a boolean success field here and get rid of the FailedMetaTransaction event